### PR TITLE
Fixes #898.  Telemetry enhancements

### LINKED ISSUE #898: [Telemetry] Track shell version and other attributes
- [x] track shell version, if possible
- [x] User Id xplat style (simplified)
- [x] Ip address (PS logs this)
- [x] Python version 
- [x] Move commands to page view, errors to events

### PR DESCRIPTION
**Pull request by BurtBiel**

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/application.py [hunk 1]
```diff
@@ -87,6 +87,7 @@ def execute(self, unexpanded_argv):
         if len(argv) == 0:
             az_subparser = self.parser.subparsers[tuple()]
             _help.show_welcome(az_subparser)
+            log_telemetry('welcome')
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/application.py]: just confirm, we do want to log telemetry even for showing welcome or help.

File: src/azure-cli-core/azure/cli/core/telemetry.py [hunk 2]
```diff
@@ -64,37 +68,89 @@ def log_event(event_name, **kwargs):
...
+    except Exception as ex: #pylint: disable=broad-except
+        # Never fail the command because of telemetry, unless debugging
+        if _debugging():
+            raise ex
+
+def _get_user_machine_id():
+    return hash(platform.node() + getpass.getuser())
+
+def _get_user_azure_id():
+    try:
```
Comment (yugangw-msft) [inline src/azure-cli-core/azure/cli/core/telemetry.py]: you probably don't need a catch here, the `log_telemetry` has a common catch cluase already, so just let is surface and we will see it when `debugging` is on

File: src/azure-cli-core/azure/cli/core/application.py (hunk 1 above)
Comment (BurtBiel) [inline src/azure-cli-core/azure/cli/core/application.py]: yep, we want to see usage patterns and if people are relying on help or welcome for information frequently

---
In reply to: [79206649](https://github.com/Azure/azure-cli/pull/910#discussion_r79206649) [](ancestors = 79206649)

File: src/azure-cli-core/azure/cli/core/telemetry.py (hunk 2 above)
Comment (BurtBiel) [inline src/azure-cli-core/azure/cli/core/telemetry.py]: If there is no active subscription, we throw a CLIError, so we want to ignore those.  If anything else goes wrong, we want to catch it in CI

---
In reply to: [79207141](https://github.com/Azure/azure-cli/pull/910#discussion_r79207141) [](ancestors = 79207141)

**Pull request merged**